futaba/cogs/welcome/prune.py

```diff
@@ -0,0 +1,139 @@
+    """
+
+    has_role = role in member.roles
+    prune = member.joined_at < prune_date
+    if has_role == should_have_role:
+        return prune
+
+    return False
+
+
+class Prune:
+    def __init__(self, bot):
+        self.bot = bot
+        self.journal = bot.get_broadcaster("/welcome")
+
+    def setup(self):
+        pass
+
+    async def prune_member(self, ctx, days):
+        """
+        Checks if a member has had the guest role for longer than the number of days specified.
+        Afterwords, will kick all members that meet that condition.
+        """
+
+        # Get guild special roles
+        roles = self.bot.sql.settings.get_special_roles(ctx.guild)
+
+        # Get the date that users that join before that have to be pruned
+        prune_date = datetime.now() - timedelta(days=days)
+
+        if roles.guest:
+            logger.info(
+                "Pruning members with role %s (%d) who joined more than %d days ago",
+                roles.guest.name,
+                roles.guest.id,
+                days,
+            )
+
+            # Get users to be pruned
+            to_be_pruned = filter(
+                lambda x: prune_filter(x, prune_date, roles.guest), ctx.guild.members
+            )
+
+        elif roles.member is None:
+            # If no member role is set the prune command should not do anything
+
+            return None
+
+        else:
+            logger.info(
+                "Pruning members without role %s (%d) who joined more than %d days ago",
+                roles.member.name,
+                roles.member.id,
+                days,
+            )
+
+            # Get users to be pruned
+            to_be_pruned = filter(
+                lambda x: prune_filter(x, prune_date, roles.member, False),
+                ctx.guild.members,
+            )
+
+        pruned = []
+
+        for member in to_be_pruned:
+            if member.top_role > ctx.me.top_role:
+                await ctx.guild.kick(
+                    member, reason=f"Pruning guests older than {days} days"
+                )
+                pruned.append(str(member))
+
+            else:
+                logger.warning("Cannnot prune member %s (%d)", member.name, member.id)
+
+        return pruned
+
+    @commands.command(name="prune", aliases=["purge"])
+    @commands.guild_only()
+    @permissions.check_mod()
+    async def prune(self, ctx, days: int = 7):
+        """
+        Prunes users that have not used the !agree command for at least the given number of days.
+        Defaults to seven days
+        """
+
+        pruned_members = await self.prune_member(ctx, days)
+
+        # Check if prune_members is none as if it is there is not member role set
+        # If there is no member role set pruning members makes no sense
+        if pruned_members is None:
+            error_message = "The server has no Member role set, so pruning the server will have no affect"
```

Comment:
```suggestion
            error_message = "The server has no Member role set, so pruning the server will have no effect"
```

Reply:
I don't remember whether we call the roles `Guest` or `guest`. Can you check other messages and make it consistent with that?